app/src/main/java/com/globant/practice/presentation/view/adapter/SubscriberAdapter.java

```diff
@@ -0,0 +1,118 @@
+
+    /**
+     * Provides the manager of the user clicks on the RecyclerView
+     */
+    public interface OnUserClickListener {
+        /**
+         * Manages the user clicks
+         *
+         * @param user User instance of the object that the user makes click
+         */
+        void onUserClick(User user);
+    }
+
+    private List<User> users;
+    private final OnUserClickListener onUserClickListener;
+
+
+    /**
+     * Setter of the users list
+     *
+     * @param users List of user
+     */
+    public SubscriberAdapter(List<User> users, OnUserClickListener onUserClickListener) {
+        this.users = users;
+        this.onUserClickListener = onUserClickListener;
+    }
+
+    /**
+     * Creates a new ViewHolder to represent a item
+     *
+     * @param parent   ViewGroup on the new View will be added
+     * @param viewType The view type of the new View
+     * @return New ViewHolder
+     */
+    @Override
+    public SubscriberViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View itemView = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.subscriber_list_item, parent, false);
+        SubscriberViewHolder listHomeViewHolder = new SubscriberViewHolder(itemView);
+        return listHomeViewHolder;
+    }
+
+    /**
+     * Updates the data of item at the position given
+     *
+     * @param holder   ViewHolder of the RecyclerView
+     * @param position Position of the item
+     */
+    @Override
+    public void onBindViewHolder(SubscriberViewHolder holder, int position) {
+        User user = users.get(position);
+        holder.blindSubscriberList(user, onUserClickListener);
+    }
+
+    /**
+     * Returns the size of the list items
+     *
+     * @return Size of the list
+     */
+    @Override
+    public int getItemCount() {
+        return users.size();
+    }
+
+    /**
+     * Inner class that manages the items of the view
+     */
+    public static class SubscriberViewHolder extends RecyclerView.ViewHolder {
+        private TextView nickname;
+        private ImageView avatar;
+
+        /**
+         * Construct method of the ListHomeHolder that initializes the view items
+         *
+         * @param itemView
+         */
+        public SubscriberViewHolder(View itemView) {
+            super(itemView);
+            nickname = (TextView) itemView.findViewById(R.id.userHomeTxt);
+            avatar = (ImageView) itemView.findViewById(R.id.userHomeImg);
+        }
+
+        /**
+         * Sets the data on the view item.
+         *
+         * @param user                User information
+         * @param onUserClickListener onUserClickListener for the user clicks
+         */
+        public void blindSubscriberList(final User user, final OnUserClickListener onUserClickListener) {
+            nickname.setText(user.getLogin());
+            itemView.setOnClickListener(new View.OnClickListener() {
```

Comment:
move this to the constructor, so you avoid N new objects of View.OnClickListener